panels/updates/cc-updates-panel.c

```diff
@@ -0,0 +1,688 @@
+  else if (strength < 80)
+    return "network-wireless-signal-good-symbolic";
+  else
+    return "network-wireless-signal-excellent-symbolic";
+}
+
+static const gchar *
+get_network_status_icon_name (CcUpdatesPanel *self,
+                              NMDevice       *device,
+                              NMAccessPoint  *ap)
+{
+  if (NM_IS_DEVICE_WIFI (device))
+    {
+      switch (nm_client_get_state (self->nm_client))
+        {
+        case NM_STATE_UNKNOWN:
+        case NM_STATE_ASLEEP:
+        case NM_STATE_DISCONNECTING:
+          return "network-wireless-offline-symbolic";
+
+        case NM_STATE_DISCONNECTED:
+          return "network-wireless-offline-symbolic";
+
+        case NM_STATE_CONNECTING:
+          return "network-wireless-acquiring-symbolic";
+
+        case NM_STATE_CONNECTED_LOCAL:
+        case NM_STATE_CONNECTED_SITE:
+          return "network-wireless-no-route-symbolic";
+
+        case NM_STATE_CONNECTED_GLOBAL:
+          return get_wifi_icon_from_strength (ap);
+        }
+    }
+  else
+    {
+      switch (nm_client_get_state (self->nm_client))
+        {
+        case NM_STATE_UNKNOWN:
+        case NM_STATE_ASLEEP:
+        case NM_STATE_DISCONNECTING:
+          return "network-wired-disconnected-symbolic";
+
+        case NM_STATE_DISCONNECTED:
+          return "network-wired-offline-symbolic";
+
+        case NM_STATE_CONNECTING:
+          return "network-wired-acquiring-symbolic";
+
+        case NM_STATE_CONNECTED_LOCAL:
+        case NM_STATE_CONNECTED_SITE:
+          return "network-wired-no-route-symbolic";
+
+        case NM_STATE_CONNECTED_GLOBAL:
+          return "network-wired-symbolic";
+        }
+    }
+
+  return "network-wireless-offline-symbolic";
+}
+
+static void
+update_active_network (CcUpdatesPanel *self)
+{
+  NMAccessPoint *ap;
+  NMConnection *connection;
+  NMDevice *device;
+  const gchar *icon_name;
+
+  get_active_connection_and_device (self, &device, &connection, &ap);
+
+  /* Cleanup previous device */
+  if (self->changed_id > 0)
+    {
+      g_source_remove (self->changed_id);
+      self->changed_id = 0;
+    }
+
+  /* Setup the new device */
+  if (g_set_object (&self->current_device, device) && device)
+    {
+      self->changed_id = g_signal_connect_swapped (device,
+                                                   "state-changed",
+                                                   G_CALLBACK (on_network_changed_cb),
+                                                   self);
+    }
+
+  /* Icon */
+  icon_name = get_network_status_icon_name (self, device, ap);
+  gtk_image_set_from_icon_name (GTK_IMAGE (self->network_status_icon), icon_name, 4);
+
+  /* Name */
+  if (ap)
+    {
+      GBytes *ssid = nm_access_point_get_ssid (ap);
+      g_autofree gchar *title = nm_utils_ssid_to_utf8 (g_bytes_get_data (ssid, NULL), g_bytes_get_size (ssid));
+
+      gtk_label_set_label (GTK_LABEL (self->network_name_label), title);
```

Comment:
Let’s leave it as it is for now. I guess the network ID isn’t completely malicious, because you would have previously chosen to connect to that network.

Reply:
Sure.